### PR TITLE
Adding 3.11 nightlies for linux PyPi

### PR DESCRIPTION
**Pull request by albanD**

### CODE REVIEW

**Comment by albanD** [comment]
Well, looks like conda and conda-build don't have 3.11 support at all.
Since all our binaries (including pypi builds) rely on conda to get a consistent environment, this will have to wait.

**Comment by albanD** [comment]
Will still try to add this to the manywheel case.
PR in builder to add python3.11 to the docker images: https://github.com/pytorch/builder/pull/1090

**Review by malfet — APPROVED** [review]
Please update the title to say that its added only for Linux

File: .github/scripts/generate_binary_build_matrix.py [hunk 1]
```diff
@@ -206,6 +206,9 @@ def generate_wheels_matrix(os: str,
         for arch_version in arches:
             gpu_arch_type = arch_type(arch_version)
             gpu_arch_version = "" if arch_version == "cpu" else arch_version
+            # Skip rocm 3.11 binaries for now as the docker image are not correct
+            if python_version == "3.11" and gpu_arch_type == "rocm":
```
Comment (jithunnair-amd) [inline .github/scripts/generate_binary_build_matrix.py:210]: @albanD Blast from the past: would you happen to remember which component didn't have the "right version" in the docker image? I'm assuming you meant the manylinux image here?

Comment (albanD) [reply .github/scripts/generate_binary_build_matrix.py:210]: IIRC What I meant is that when I modified the builder repo which biulds the docker images, I only updated the linux docker images for cuda.

Comment (jithunnair-amd) [reply .github/scripts/generate_binary_build_matrix.py:210]: Ah okay, thanks!

**Pull request closed without merging**